### ISSUE TITLE
CMake 3.10: POLICY CMP0071 warnings

### ISSUE THREAD

**Issue by annulen** [annulen/webkit#638]
cmake 3.10 wants to apply automoc to generated files, and wants us to set CMP0071 to OLD or NEW, otherwise lots of warnings are printed. Some of them are caused by files generated in WebKit, others by macros from Qt which will hopefully be fixed in new Qt versions.

Problem with setting policy is that we have multiple `cmake_minimum_required` calls, I think it needs to be fixed in upstream as well

```diff
diff --git a/Source/WebKit2/CMakeLists.txt b/Source/WebKit2/CMakeLists.txt
index aa7863cbd27..ae9887dc133 100644
--- a/Source/WebKit2/CMakeLists.txt
+++ b/Source/WebKit2/CMakeLists.txt
@@ -1,6 +1,10 @@
 include(CheckCXXSourceCompiles)
 include(CheckFunctionExists)
 
+if (PORT STREQUAL "Qt" AND POLICY CMP0071)
+    cmake_policy(SET CMP0071 OLD)
+endif ()
+
 set(WebKit2_INCLUDE_DIRECTORIES
     "${JAVASCRIPTCORE_DIR}/llint"
     "${WEBKIT2_DIR}"
```

**Comment by antonio-rojas** [comment]
Not only getting warnings here, but actual errors:

CMake Error in Source/WebKit/CMakeLists.txt:
  Cannot find source file:

    qt/WidgetApi/qwebfullscreenrequest.cpp

  Tried extensions .c .C .c++ .cc .cpp .cxx .m .M .mm .h .hh .h++ .hm .hpp
  .hxx .in .txx

**Comment by annulen** [comment]
See f51554bf104ab0491370f66631fe46143a23d5c2

**Comment by annulen** [comment]
this is mostly fixed after 8cfb3f2f158ff82ad37f6addf98131dab446e2d3
